### PR TITLE
Proof Generation returns the concrete proof instead of a deserialized one

### PR DESCRIPTION
**Pull request by ilitteri**

### CODE REVIEW

File: src/marlin/mod.rs [hunk 1]
```diff
@@ -64,16 +63,9 @@ pub fn generate_proof(
...
     rng: &mut StdRng,
-) -> Result<Vec<u8>> {
-    let proof = MarlinInst::prove_from_constraint_system(&proving_key, constraint_system, rng)
-        .map_err(|_e| anyhow!("Error in prove_from_constraint_system"))?;
-
-    let mut bytes_proof = Vec::new();
-    proof
-        .serialize(&mut bytes_proof)
-        .map_err(|_e| anyhow!("Error serializing proof"))?;
-
```
Comment (pablodeymo) [inline src/marlin/mod.rs:75]: I would suggest to add a new method to have serialization and deserialization

Comment (ilitteri) [reply src/marlin/mod.rs:75]: Done!

Comment (pablodeymo) [reply src/marlin/mod.rs:75]: Great

**Review by pablodeymo — APPROVED** [review]
LGTM

**Pull request merged**